content/en/blog/2025/opentelemetry-for-legacy-app/index.md

```diff
@@ -0,0 +1,148 @@
+
+**Step 1: Set up the Environment**
+
+In our terminal, we'll configure the agent without touching the application itself.
+
+```bash
+# --- Part 1: Basic System Health Configuration ---
+
+# 1. Give our service a descriptive name
+export OTEL_SERVICE_NAME=legacy-part-processor
+
+# 2. Enable ONLY the metrics exporter
+export OTEL_METRICS_EXPORTER=otlp
+
+# 3. Point the agent to the collector's gRPC endpoint
+export OTEL_EXPORTER_OTLP_ENDPOINT=http://127.0.0.1:4317
+
+# 4. Specify OTLP protocol
+export OTEL_EXPORTER_OTLP_PROTOCOL=grpc
+
+# 5. Enable runtime metrics for Java 8
+export OTEL_INSTRUMENTATION_RUNTIME_TELEMETRY_JAVA8_ENABLED=true
+
+# 6. Attach the OpenTelemetry Java agent
+export _JAVA_OPTIONS="-javaagent:./opentelemetry-javaagent.jar"
+```
+
+**Step 2: Run the Unmodified Application and confirm it's working**
+
+```bash
+./legacy_app
+```
+
+First, let's verify that our legacy application is running. A quick look at the terminal logs shows a steady stream of messages, confirming that our process is alive and well:
+
+```bash
+[C Wrapper] Reading new Part ID from assembly line: 3035
+[Java Processor] Received Part ID 3035. Fetching processing parameters...
+[Java Processor] Part ID 3035 processed successfully.
+[C Wrapper] Processing request for Part ID 3035 completed successfully.
+
+[C Wrapper] Reading new Part ID from assembly line: 3036
+[Java Processor] Received Part ID 3036. Fetching processing parameters...
+[Java Processor] Part ID 3036 processed successfully.
+[C Wrapper] Processing request for Part ID 3036 completed successfully.
+
+[C Wrapper] Reading new Part ID from assembly line: 3037
+[Java Processor] Received Part ID 3037. Fetching processing parameters...
+[Java Processor] Part ID 3037 processed successfully.
+[C Wrapper] Processing request for Part ID 3037 completed successfully.
+```
+
+The data from our application follows a standard path: the OTel agent sends metrics to the OpenTelemetry Collector, which are then scraped by Prometheus for storage. We then use Grafana to connect to Prometheus and visualize the data. Since this is a common and well-documented setup, we'll skip the specific configuration files and jump right into what this visibility allows us to see.
+
+**Step 3: Build Basic Monitoring Dashboard**
+
+This is the moment it all comes together. We can now head over to Grafana, and with just three simple queries, instantly build our first dashboard. And just like that, we have basic monitoring for our application, covering average CPU utilization, average memory consumption, and time spent in garbage collection.
+
+![Basic Monitoring Dashboard](basic-monitoring.png)
```

Comment:
It looks like the dashboard images were taken in Grafana Cloud, which isn't open source. Please crop the side nav out of both images. Thanks!

Reply:
Sure, just cropped it. It is now fixed in latest commit.